src/setfit/modeling.py

```diff
@@ -633,30 +634,87 @@ def forward(self, sentence_features, labels=None, mask=None):
         return loss
 
 
-def sentence_pairs_generation(sentences, labels, pairs):
-    # Initialize two empty lists to hold the (sentence, sentence) pairs and
-    # labels to indicate if a pair is positive or negative
+def positive_sentence_pairs_generate(
+    sentences: np.ndarray[str], labels: np.ndarray[int], max_pairs: int, unique_pairs: bool
```

Comment:
```suggestion
    sentences: "np.ndarray[str]", labels: "np.ndarray[int]", max_pairs: int, unique_pairs: bool
```
Alternatively, `ArrayLike` may be applicable? https://numpy.org/doc/stable/reference/typing.html#numpy.typing.ArrayLike

Otherwise we'll run into errors (as you can see with the tests)

Reply:
Ahh yes. I'm happy with those changes or just back to `np.ndarray` only and will add the array-types in the docstring? 
The `npt.ArrayLike` loads stuff pretty ugly and I've not seen in other projects before...

Reply:
I also have never used `npt.ArrayLike` before. I think `"np.ndarray[int]"` is more than fine for now.